core/src/raw/oio/stream/api.rs

```diff
@@ -172,3 +195,55 @@ where
         Pin::new(this.inner).poll_reset(cx)
     }
 }
+
+/// Stream for the [`chain`](StreamExt::chain) method.
+#[must_use = "streams do nothing unless polled"]
+pub struct Chain<S1: Stream, S2: Stream> {
+    first: Option<S1>,
+    second: S2,
+}
```

Comment:
Why not name its properties like a link-list node?

Reply:
Inspired by futures::Chain.

Reply:
🤦Ok, I see.